packages/react/src/tabs/tab/useTabsTab.ts

```diff
@@ -75,32 +76,56 @@ function useTabsTab(parameters: useTabsTab.Parameters): useTabsTab.ReturnValue {
 
   const tabPanelId = index > -1 ? getTabPanelIdByTabValueOrIndex(valueParam, index) : undefined;
 
+  const isPressingRef = React.useRef(false);
+  const isPrimaryButtonRef = React.useRef(false);
+
   const getRootProps = React.useCallback(
     (externalProps = {}) => {
       return mergeReactProps<'button'>(
         externalProps,
-        mergeReactProps<'button'>(
-          {
-            role: 'tab',
-            'aria-controls': tabPanelId,
-            'aria-selected': selected,
-            id,
-            ref: handleRef,
-            onClick(event) {
+        {
+          role: 'tab',
+          'aria-controls': tabPanelId,
+          'aria-selected': selected,
+          id,
+          ref: handleRef,
+          onClick(event) {
+            if (selected) {
+              return;
+            }
+
+            onTabActivation(tabValue, event.nativeEvent);
+          },
+          onFocus(event) {
+            if (!activateOnFocus || selected) {
+              return;
+            }
+
+            if (!isPressingRef.current || (isPressingRef.current && isPrimaryButtonRef.current)) {
               onTabActivation(tabValue, event.nativeEvent);
-            },
-            onFocus(event) {
-              if (!activateOnFocus) {
-                return;
-              }
-
-              if (selectedTabValue !== tabValue) {
-                onTabActivation(tabValue, event.nativeEvent);
-              }
-            },
+            }
+          },
+          onPointerDown(event) {
+            if (selected) {
+              return;
+            }
+
+            isPressingRef.current = true;
+
+            function handlePointerUp() {
+              isPressingRef.current = false;
+              isPrimaryButtonRef.current = false;
+            }
+
+            if (!event.button || event.button === 0) {
```

Comment:
We use

```suggestion
            if (event.button === 0) {
```

in all Material UI codebase I have never seen an issue with `event.button` being missing. Can't we reduce bundle size here?

Reply:
Tried removing this check in NumberField [here](https://github.com/mui/base-ui/pull/1318/commits/6e63c2bf352e0cb382a1f6ac7b89a35534e1b961) too but without it a test fails: https://app.circleci.com/pipelines/github/mui/base-ui/7042/workflows/cc6f84c4-5cc2-488c-b741-ebb30c275d4a/jobs/61996/parallel-runs/0/steps/0-106

Not sure if it's just a test issue but I'd prefer to keep the main button check the same for now CC @atomiks